server/service/osquery_utils/queries.go

```diff
@@ -1300,71 +1294,79 @@ func directIngestChromeProfiles(ctx context.Context, logger log.Logger, host *fl
 	return ds.ReplaceHostDeviceMapping(ctx, host.ID, mapping, fleet.DeviceMappingGoogleChromeProfiles)
 }
 
+// directIngestBattery ingests battery data from a host on a Windows or macOS platform
+// and calculates the battery health based on cycle count and capacity.
+// Due to a known osquery issue with M1 Macs (https://github.com/fleetdm/fleet/issues/6763)
+// and the ommission of the `health` column on Windows, we are not leveraging the `health`
+// column in the query and instead aligning the definition of battery health with between
+// macOS and Windows.
 func directIngestBattery(ctx context.Context, logger log.Logger, host *fleet.Host, ds fleet.Datastore, rows []map[string]string) error {
 	mapping := make([]*fleet.HostBattery, 0, len(rows))
+
 	for _, row := range rows {
-		cycleCount, err := strconv.Atoi(EmptyToZero(row["cycle_count"]))
+		health, cycleCount, err := generateBatteryHealth(row, logger)
 		if err != nil {
-			return err
+			level.Error(logger).Log("op", "directIngestBattery", "hostID", host.ID, "err", err)
 		}
 
-		switch host.Platform {
-		case "darwin":
-			mapping = append(mapping, &fleet.HostBattery{
-				HostID:       host.ID,
-				SerialNumber: row["serial_number"],
-				CycleCount:   cycleCount,
-				// database type is VARCHAR(40) and since there isn't a
-				// canonical list of strings we can get for health, we
-				// truncate the value just in case.
-				Health: fmt.Sprintf("%.40s", row["health"]),
-			})
-		case "windows":
-			health, err := generateWindowsBatteryHealth(row["designed_capacity"], row["max_capacity"])
-			if err != nil {
-				level.Error(logger).Log("op", "directIngestBattery", "hostID", host.ID, "err", err)
-			}
-
-			mapping = append(mapping, &fleet.HostBattery{
-				HostID:       host.ID,
-				SerialNumber: row["serial_number"],
-				CycleCount:   cycleCount,
-				Health:       health,
-			})
-		}
+		mapping = append(mapping, &fleet.HostBattery{
+			HostID:       host.ID,
+			SerialNumber: row["serial_number"],
+			CycleCount:   cycleCount,
+			Health:       health,
+		})
 	}
+
 	return ds.ReplaceHostBatteries(ctx, host.ID, mapping)
 }
 
 const (
-	batteryStatusUnknown     = "Unknown"
-	batteryStatusDegraded    = "Check Battery"
-	batteryStatusGood        = "Good"
-	batteryDegradedThreshold = 80
+	batteryStatusUnknown      = "Unknown"
+	batteryStatusDegraded     = "Service recommended"
+	batteryStatusGood         = "Normal"
+	batteryDegradedThreshold  = 80
+	batteryDegradedCycleCount = 1000
```

Comment:
Do we want the cycle count in here if the battery is still >= 80% of design capacity? Link me if there's a product decision around this.

Reply:
recently updated in the figma https://www.figma.com/design/P5lBL6cRfDt7DgE4y0ciiZ/%2319619-Show-windows-battery-health?node-id=1-112&node-type=canvas&t=V11ATANuFvmJlVIf-0